### ISSUE TITLE
Radare2 Update?

### ISSUE THREAD

**Issue by geeksailor** [REMnux/salt-states#274]
Devs,
While I was running an update and I ran into a issue with what I believe was Radare2 updating. When I looked through the salt to look more into the program I noticed that it has `set version = '5.6.8'` hardcoded into it. However when I went to their Git, the most current version is now up to 5.8.8 (Jun 2023) and the version hardcoded into the salt is an update from April 2022. Just wanted to let you all know in case you want to possibly push an update to a more current version of the program.
Cheers!

EDIT: Don't forget to update the hardcoded SHA256 as well if you update the version. I stupidly forgot to update that when trying to test running an update to that stack on my local machine and could not figure out for the life of me why it was failing.

**Comment by geeksailor** [comment]
I'm not sure what is going on here but I'm assuming that because the version in `/usr/local/src/` is not similar to the most current version it is rejecting it as a downgrade without accepting the downgrade and there for failing?
[saltstack.log](https://github.com/REMnux/salt-states/files/14057116/saltstack.log)

I grabbed the current version (5.8.8) to where it says it stores the files, updated the version number and the hash to the radare2.sls, stuffed it into the gzip file locally on under my 2024.2.2 folder, updated the sha256 file, and reran `remnux upgrade` and worked with no issues this time. So that seemed to fix it for me. 

Since my problem is fixed, I'll close this thread out as soon as someone weighs in on the update and whether or the devs decide to go ahead on it.
Cheers!

**Comment by digitalsleuth** [comment]
Hi @geeksailor The reason for this error is because the version you had installed during the process was 5.8.8, but the SaltStack installation had wanted to install 5.6.8, and this technically is considered a downgrade. As such, in order for this to work properly, the "allow-downgrade" option would have to be set, which it's not. We do not have it set in the state because we don't want users to have to downgrade in order to use REMnux. Additionally, there is no native option in SaltStack to set the "allow-downgrade" flag to the command.

However, since you've identified the version as being out of date, I'll submit a PR for an updated version of it this weekend.

Cheers!

**Comment by digitalsleuth** [comment]
Hi @geeksailor Just checking in to see if your installation was successful after the [PR for Radare2](https://github.com/REMnux/salt-states/pull/276) was released.

**Comment by geeksailor** [comment]
Hey @digitalsleuth,
Sorry I kinda disappeared. Works picked up quite a bit and been keeping my busier than usual which is good. But the downside being that I've barely had any time as of lately to test or play with Remnux for the past few months. I do think I played around with it some more but I can not remember if the install was successful or not. I'll break out my copy this weekend and check and see if I did in fact work and let you know.
Cheers!
